### DISCUSSION TITLE
wrong usage of serializer_class attr?

### DISCUSSION THREAD

**Discussion by jokiefer** [django-json-api/django-rest-framework-json-api#1015] [Q&A] [answered]
Hi folks,

I stumbled across the following [line](https://github.com/django-json-api/django-rest-framework-json-api/blob/ffedc6bb755f9b81af7ee423ba64021535fef96c/rest_framework_json_api/parsers.py#L163) and i'am not sure if this is the right way to figure out the `serializer_class` of the view, cause the `ModelViewSet` gets his own `serializer_class` by calling the `get_serializer_class` function. 

Should this better rewrite to: 
```python

get_serializer_func = getattr(view, "get_serializer_class", None)
if get_serializer_func:
   
    serializer_class = get_serializer_func()
else:
   serializer_class = None

```

**Comment by sliverc** [comment, marked as answer]
This code bit was an invalid bug fix to allow the usage of the field name `type` again. The JSON:API spec does not allow this though so this code block will be removed in next major version (see [TODO](https://github.com/django-json-api/django-rest-framework-json-api/blob/master/rest_framework_json_api/parsers.py#L165)). See #993 for more details.
